removes subjective sadness from robot surveyor module

Ninetailed opened the pull request:
The surveyor module for the flying robot chassis was broken or at least somewhat annoying to use in a number of ways. This should hopefully improve that.

🆑 Ninetailed
rscadd: Robots can now unbuckle themselves and others from things by clicking on them, just like a real boy!
rscadd: Robots with the GPS/RPD as an active module now get coordinates in the Status box the same as humans do
rscadd: Surveyor module now has a harvester to gather plants for the bioreactor
tweak: Robot net gun no longer has a safety, as selecting the module for use serves the same purpose
tweak: Vines can no longer grab you if you're flying
bugfix: Net gun now actually fires nets instead of dropping them
bugfix: Robot net gun now spawns with shells loaded
bugfix: All living mobs, including robots, can break free of vines using the Resist verb, rather than this being restricted to carbon life
/🆑

Fixes #25310

Ninetailed committed:
Robot/Surveyor module quality of life changes

The surveyor module for the flying robot chassis was broken or at least somewhat annoying to use in a number of ways. This should hopefully improve that.

Changes:

- rscadd: Robots can now unbuckle themselves and others from things by clicking on them, just like a real boy!
- rscadd: Robots with the GPS/RPD as an active module now get coordinates in the Status box the same as humans do
- rscadd: Surveyor module now has a harvester to gather plants for the bioreactor
- tweak: Robot net gun no longer has a safety, as selecting the module for use serves the same purpose
- tweak: Vines can no longer grab you if you're flying
- bugfix: Net gun now actually fires nets instead of dropping them
- bugfix: Robot net gun now spawns with shells loaded
- bugfix: All living mobs, including robots, can break free of vines using the Resist verb, rather than this being restricted to carbon life

Fixes #25310

Ninetailed commented:
Despite the auto-tagger thinking it is, this isn't actually a map change, it's just flagged up as one because the surveyor module is in with the Torch map files.

SierraKomodo commented:
> Despite the auto-tagger thinking it is, this isn't actually a map change, it's just flagged up as one because the surveyor module is in with the Torch map files.

I believe the map flag is meant to indicate changes to a map-specific item/file, not necessarily .dmm changes.

SierraKomodo commented on code/modules/hydroponics/spreading/spreading_response.dm:
```diff
@@ -71,6 +71,10 @@
 		return
 
+	if(victim.is_floating)
+		victim.visible_message("<span class='danger'>Tendrils lash out towards \the [victim] but \the [src] can't quite reach them as they float above!</span>", "<span class='danger'>Tendrils lash out from \the [src] below but can't quite reach you!</span>")
```
SPAN_ classes. Also, I'd suggest using warning instead of danger - It's not actually grabbing so it's not an immediate danger.

```suggestion
		victim.visible_message(
			SPAN_WARNING("Tendrils lash out towards \the [victim] but \the [src] can't quite reach them as they float above!"),
			SPAN_WARNING("Tendrils lash out from \the [src] below but can't quite reach you!")
		)
```

Ninetailed replied:
I was going for consistency with what was already there, but sure, easy enough to change.

SierraKomodo replied:
We're taking a 'Update it as things are added/changes' approach with span classes. One of these days when I feel masochistic enough I'll probably PR a mass update.

Ninetailed replied:
I'll adopt the same approach with my future changes.

SierraKomodo commented on maps/torch/robot/module_flying_surveyor.dm:
```diff
@@ -50,13 +51,17 @@
 	. = ..()
 	for(var/flag_type in flag_types)
 		equipment += new flag_type(src)
+	// Net gun should start fully loaded
+	var/obj/item/weapon/gun/launcher/net/borg/gun = locate() in equipment
+	while(LAZYLEN(gun.shells) < gun.max_shells)
+		gun.load(new /obj/item/weapon/net_shell)
```
I think this is better off put into the `Initialize()` call for the borg's gun itself.

Ninetailed replied:
Fair point. I can't remember what my reasoning for putting it here was in the first place.

Ninetailed committed:
Refactor spreading_response to use SPAN_ defines

Also adjusted some of the message severity levels, as the vines are being treated as much more alarming than they actually are when they fail to grab you.

SierraKomodo reviewed (changes requested):
Everything seems to be working as described - However, I'm not seeing a way to be able to reload the net gun as a surveyor borg.

Ninetailed commented:
> 
> 
> Everything seems to be working as described - However, I'm not seeing a way to be able to reload the net gun as a surveyor borg.

There were actually two ways already to fill the empty net gun before this PR. First, consumables are refilled when the borg is in a charger - beacons as well as net shells. The code for that is in the surveyor module itself, which might be why I initially put the "start loaded" code in there too. Second, as a backup, you can click shells in the world with the net gun to load them in.

That second one actually doesn't work with the net shells in the Charon because those are for a different class of weapon – the explorers' net gun is really just a shotgun dressed up as something else, while the borg's is a dedicated net gun that can't take other ammo. Might be worth making them compatible some day, but since the charger does the job, I wasn't too worried about it.

SierraKomodo commented:
> > Everything seems to be working as described - However, I'm not seeing a way to be able to reload the net gun as a surveyor borg.
> 
> There were actually two ways already to fill the empty net gun before this PR. First, consumables are refilled when the borg is in a charger - beacons as well as net shells. The code for that is in the surveyor module itself, which might be why I initially put the "start loaded" code in there too. Second, as a backup, you can click shells in the world with the net gun to load them in.
> 
> That second one actually doesn't work with the net shells in the Charon because those are for a different class of weapon – the explorers' net gun is really just a shotgun dressed up as something else, while the borg's is a dedicated net gun that can't take other ammo. Might be worth making them compatible some day, but since the charger does the job, I wasn't too worried about it.

Ah, that works then - I've gotten spoiled by medborg where things auto-recharge outside of the charger and just drain your cell while they do it and didn't think to try charging.

Spookerton commented:
Not sure I'm happy about everything that floats being immune to vines, but we'll see how that goes.

Ninetailed commented:
> Not sure I'm happy about everything that floats being immune to vines, but we'll see how that goes.

That's the change I was least sure about myself. I won't be offended if you decide to take it back out.
